test/tmetrics.m

```diff
@@ -537,6 +537,40 @@ function testHistogramDelta(testCase)
                 verifyEqual(testCase, str2double(counts{len}), sum(currentvals>bounds(len-1)));
             end
         end
+
+        function testGetSetMeterProvider(testCase)
+            % testGetSetMeterProvider: setting and getting global instance of MeterProvider
+            exporter = opentelemetry.exporters.otlp.OtlpHttpMetricExporter(...
+                "PreferredAggregationTemporality", "Delta");
```

Comment:
I don't think you need "Delta" temporality here. I would just call the exporter constructor with no input and use the default temporality.

Reply:
For some reason after implementing all these changes and running the tests, all the tracer tests which are run after the metrics tests fail. I added the Delta temporality back in, and the tracer tests are now passing again. Do you know why that could be happening?

Reply:
How did the tests fail? What are the error messages?

Reply:
It was Unrecognized field name "resourceSpans" happening on all the tracer tests. Not sure if it's related to this though. Now when I run the tests I get this error when running tcontectPropogation: [Error] File: C:\Users\dnarula\OneDrive - MathWorks\Documents\OpenTelemetry-Matlab\build\otel-cpp\src\opentelemetry-cpp\exporters\otlp\src\otlp_http_exporter.cc:123 [OTLP HTTP Client] ERROR: Export 1 trace span(s) error: 1

It seems the behavior of the tests are very inconsistent, somtimes they pass and sometimes they fail.

Reply:
The resourceSpan error is unrelated. I think currently it is a sporadic error depending on the run. The reason why it is happening is that even when we do "clear mp" before readJsonResults, we are still not stopping the MeterProvider from regularly exporting. The MeterProvider mp contains a shared pointer that points to the cpp MeterProvider object. But that cpp MeterProvider object is also pointed to by the global MeterProvider with a shared pointer. As long as there is at least one shared pointer pointing to it, the cpp MeterProvider is not getting destroyed. 
After the metric tests are completed and when the tracing tests are running, the collector is getting data from both metrics and tracing. Depending on the timing, the tracing tests sometimes pick up the metric results, and the metric results doesn't have the resourceSpan field. 
So we need to figure out how to deal with this larger issue. But that can be after your PR gets accepted. I would recommend you can continue to remove the "delta" temporality. If the tests fail, we can ignore for now, and deal with it after.

Reply:
Ok sounds good thanks. I deleted my earlier comment about the error with tcontextPropagation because that was just caused by a typo so that's fine.

I just confirmed by testing a few times that the resourceSpan error does seem to occur when the "delta" temporality is removed, and the error goes away when I add it back. I'm not sure what the connection is there, but I can still remove the "delta" temporality if you'd like.